clippy_lints/src/casts/cast_possible_truncation.rs

```diff
@@ -199,3 +229,20 @@ fn offer_suggestion(
         SuggestionStyle::ShowAlways,
     );
 }
+
+// FIXME(@tesuji): May extend this to a validator functions to include:
+// * some ABI-guaranteed STD types,
+// * some non-local crate types suggested in [PR-12962][1].
+// [1]: https://github.com/rust-lang/rust-clippy/pull/12962#discussion_r1661500351
+fn is_valid_sizeof(ty: Ty<'_>) -> bool {
+    if ty.is_primitive_ty() || ty.is_any_ptr() || ty.is_box() || ty.is_slice() {
+        return true;
+    }
+    if let ty::Adt(def, ..) = ty.kind()
+        && def.did().is_local()
```

Comment:
This isn't enough of a constraint. The types of all the fields to to be validated as well.

Reply:
Should be fixed by new commits.